include/internal/hashtable.h

```diff
@@ -198,8 +218,11 @@ pfx ossl_unused int ossl_ht_##name##_##vtype##_insert(HT *h, HT_KEY *key,      \
     inval.value = data;                                                        \
     inval.type_id = &name##_##vtype##_id;                                      \
     rc = ossl_ht_insert(h, key, &inval, olddata == NULL ? NULL : &oval);       \
-    if (oval != NULL)                                                          \
+    if (oval != NULL) {                                                        \
         *olddata = (vtype *)oval->value;                                       \
```

Comment:
I'm not sure about this. at line 220 we use ternary operator `odlddata == NULL ? ....` if olddata is NULL we use pass &oval (a local variable instead).

here at line 222 we do potential NULL pointer dereference. I think we should add `if (olddata != NULL)` at least